codegen/src/TypeMapper.scala

```diff
@@ -1,35 +1,69 @@
 package besom.codegen
 
+import besom.codegen.Config.ProviderConfig
+
 import scala.util.matching.Regex
 import scala.meta._
 import scala.meta.dialects.Scala33
 import besom.codegen.metaschema._
 
+import scala.annotation.unused
+
+case class TypeToken private (provider: String, module: String, `type`: String, @unused dummy: Boolean = false) {
+  def asString: String = s"${provider}:${module}:${`type`}"
+}
+
+object TypeToken {
+  private val typeTokenFmt: String    = "(.*):(.*)?:(.*)" // provider:module:type
```

Comment:
`?` seems redundant here as `.*` would accept an empty string anyway

Reply:
yes, you're probably right, but this one better states the intention IMO